### PR TITLE
prov/efa: Fix shm initialization path on error

### PR DESCRIPTION
**Pull request by wckzhang**
Signed-off-by: William Zhang <wilzhang@amazon.com>

### CODE REVIEW

File: prov/efa/src/efa_user_info.c [hunk 1]
```diff
@@ -521,27 +521,38 @@ int efa_getinfo(uint32_t version, const char *node,
 		shm_info_initialized = true;
 	}
 
-	if (hints && hints->ep_attr && hints->ep_attr->type == FI_EP_DGRAM)
-		return efa_user_info_get_dgram(version, node, service, flags, hints, info);
+	if (hints && hints->ep_attr && hints->ep_attr->type == FI_EP_DGRAM) {
+		err = efa_user_info_get_dgram(version, node, service, flags, hints, info);
+		if (err)
+			goto error;
+		else
```
Comment (wzamazon) [inline prov/efa/src/efa_user_info.c:528]: It seems we do not need `else` and we can `return 0`.

Comment (wckzhang) [reply prov/efa/src/efa_user_info.c:528]: That's mostly just style right? Do you want it changed or is it okay as is?

Comment (wzamazon) [reply prov/efa/src/efa_user_info.c:528]: I will prefer the change if it is not too much trouble.

**Comment by shijin-aws** [comment]
How far do we need to backport?

**Comment by wzamazon** [comment]
>How far do we need to backport?

Definitely 1.16.x

File: prov/efa/src/efa_user_info.c (hunk 1 above)
Comment (wckzhang) [reply prov/efa/src/efa_user_info.c]: Done

**Review by wzamazon — APPROVED** [review]

**Comment by wzamazon** [comment]
Can you please backport to v1.16.x?

**Pull request merged**